### PR TITLE
Check remaining size on fast heap

### LINKED ISSUE #183: Add FastHeapRemaining to SystemStats
Currently, we don't have a way to see how much of the fast heap we are using. `SystemStats` should be extended to have an additional entry (e.g. `FastHeapRemaining`) to show how much memory is still available.

### PR DESCRIPTION
**Pull request by tracernz**
Also adds descriptions to SystemStats UAVO fields while I was in there.

Tested on Quanton.
To test:
- [x] F1
- [x] F3/4 FreeRTOS

Possible that the PIOS_SYSTEM_STACK_SIZE may need bumped on F1s.

Fixes #183

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/235)

### CODE REVIEW

File: flight/PiOS/Common/pios_heap.c [hunk 1]
```diff
@@ -202,6 +202,33 @@ size_t PIOS_heap_get_free_size(void)
 	return free_bytes;
 }
 
+
+#if defined(PIOS_INCLUDE_FASTHEAP)
+
+size_t PIOS_fastheap_get_free_size(void)
+{
+#if defined(PIOS_INCLUDE_FREERTOS)
```
Comment (tracernz) [inline flight/PiOS/Common/pios_heap.c]: This could be a lot cleaner if we drop FreeRTOS. ;-)

**Comment by mlyle** [comment]
Breaks sim:

```
/var/lib/jenkins/jobs/dron/workspace@3/Nodes/master/build/sim_posix/systemmod.o: In function `updateStats':
/var/lib/jenkins/jobs/dron/workspace@3/Nodes/master/flight/Modules/System/systemmod.c:567: undefined reference to `PIOS_fastheap_get_free_size'
```

**Comment by tracernz** [comment]
Thanks, edited the original commit so it doesn't break the build.

**Comment by mlyle** [comment]
Reviewed 4 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[flight/PiOS/Common/pios_heap.c, line 210 \[r1\]](https://reviewable.io:443/reviews/d-ronin/dronin/235#-K5EP4jBUQooc4-OVWUP-r1-210:-K5F8GL-8HPf4iiCKyBM:5173050) ([raw file](https://github.com/d-ronin/dronin/blob/77ba3fd7a9472f2b35ffdd89e3bdd148d0559250/flight/PiOS/Common/pios_heap.c#L210)):*
;P

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/235)*

**Comment by mlyle** [comment]
:lgtm_strong: 

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/235#-:-K5F8WHtTBBgHsQC5jeU:-1868698381)*

**Comment by tracernz** [comment]
Tested on CC3D, 120 bytes remaining in System task (compiled with DIAG_TASKS=YES).

**Comment by tracernz** [comment]
Also good bench test on Revo (FreeRTOS).

**Comment by mluessi** [comment]
LGTM, will test on Brain (F4, ChibiOS)

**Pull request merged**